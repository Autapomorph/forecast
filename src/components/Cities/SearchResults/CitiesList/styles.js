import styled from 'styled-components';

export const StyledCitiesList = styled.ul`
  padding: 0;
  margin-bottom: 15px;
  list-style: none;
`;

export const StyledCitiesListItem = styled.li`
  margin-bottom: 15px;
  border: 1px solid var(--light-gray);
  border-radius: 5px;
  background: var(--white);
  color: var(--contrast-text-color);
  font-weight: bold;
`;
