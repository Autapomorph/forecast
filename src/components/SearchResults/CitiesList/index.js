import React from 'react';

import CitiesItem from '../CitiesItem';

const CitiesList = ({
  cities,
  checkIfFeatured,
  fetchCity,
  addCityToFeatured,
  removeCityFromFeatured,
}) => (
  <div>
    <h3>SEARCH RESULTS</h3>
    <ul>
      {Object.values(cities).map(city => (
        <li key={city.id}>
          <CitiesItem
            city={city}
            isFeatured={checkIfFeatured(city.id)}
            fetchCity={fetchCity}
            addCityToFeatured={addCityToFeatured}
            removeCityFromFeatured={removeCityFromFeatured}
          />
        </li>
      ))}
    </ul>
  </div>
);

export default CitiesList;
