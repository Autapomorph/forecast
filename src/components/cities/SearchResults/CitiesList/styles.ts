import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledCitiesList = styled.ul`
  padding: 0;
  margin: 12px 0 0;
  list-style: none;
`;
