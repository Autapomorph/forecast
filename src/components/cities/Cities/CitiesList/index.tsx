import React from 'react';

import { City, Coords } from 'models';
import CitiesItem from '../CitiesItem';

import * as S from './styles';

type Props = {
  cities: City[];
  checkIfFeatured: (cityId: City['id']) => boolean;
  fetchCity: (cityCoords: Coords) => void;
  addToFeatured: (city: City) => void;
  removeFromFeatured: (cityId: City['id']) => void;
};

const CitiesList = ({
  cities,
  checkIfFeatured,
  fetchCity,
  addToFeatured,
  removeFromFeatured,
}: Props): React.ReactElement => (
  <S.CitiesList>
    {Object.values(cities).map(city => (
      <CitiesItem
        key={city.id}
        city={city}
        isFeatured={checkIfFeatured(city.id)}
        fetchCity={fetchCity}
        addToFeatured={addToFeatured}
        removeFromFeatured={removeFromFeatured}
      />
    ))}
  </S.CitiesList>
);

export default CitiesList;
