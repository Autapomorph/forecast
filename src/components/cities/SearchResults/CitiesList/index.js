import React from 'react';

import CitiesItem from '../CitiesItem';

import { StyledCitiesList } from './styles';

const CitiesList = ({
  cities,
  checkIfFeatured,
  fetchCityWeather,
  addCityToFeatured,
  removeCityFromFeatured,
}) => (
  <StyledCitiesList>
    {Object.values(cities).map(city => (
      <CitiesItem
        key={city.id}
        city={city}
        isFeatured={checkIfFeatured(city.id)}
        fetchCityWeather={fetchCityWeather}
        addCityToFeatured={addCityToFeatured}
        removeCityFromFeatured={removeCityFromFeatured}
      />
    ))}
  </StyledCitiesList>
);

export default CitiesList;
