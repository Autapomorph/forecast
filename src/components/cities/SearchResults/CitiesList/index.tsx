import React from 'react';

import { City, Coords } from 'models';
import CitiesItem from '../CitiesItem';

import { StyledCitiesList } from './styles';

type Props = {
  cities: City[];
  checkIfFeatured: (cityId: City['id']) => boolean;
  fetchCity: (cityCoords: Coords) => void;
  addCityToFeatured: (city: City) => void;
  removeCityFromFeatured: (cityId: City['id']) => void;
};

const CitiesList: React.FC<Props> = ({
  cities,
  checkIfFeatured,
  fetchCity,
  addCityToFeatured,
  removeCityFromFeatured,
}): React.ReactElement => (
  <StyledCitiesList>
    {Object.values(cities).map(city => (
      <CitiesItem
        key={city.id}
        city={city}
        isFeatured={checkIfFeatured(city.id)}
        fetchCity={fetchCity}
        addCityToFeatured={addCityToFeatured}
        removeCityFromFeatured={removeCityFromFeatured}
      />
    ))}
  </StyledCitiesList>
);

export default CitiesList;
