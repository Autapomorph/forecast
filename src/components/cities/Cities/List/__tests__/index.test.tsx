import React from 'react';
import { shallow } from 'enzyme';

import CitiesList from '..';

describe('CitiesList tests', () => {
  const props = {
    cities: [
      {
        id: 0,
        name: 'name',
        region: 'region',
        countryName: 'countryName',
        countryCode: 'countryCode',
        coords: { latitude: 0, longitude: 0 },
      },
      {
        id: 1,
        name: 'name2',
        region: 'region2',
        countryName: 'countryName2',
        countryCode: 'countryCode2',
        coords: { latitude: 0, longitude: 0 },
      },
    ],
    checkIfFeatured: jest.fn(),
    fetchCity: jest.fn(),
    addToFeatured: jest.fn(),
    removeFromFeatured: jest.fn(),
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders properly', () => {
    const wrapper = shallow(<CitiesList {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render CitiesItem', () => {
    const wrapper = shallow(<CitiesList {...props} />);
    const items = wrapper.find('CitiesItem');

    expect(items).toHaveLength(2);
    expect(items.at(0).key()).toBe('0');
    expect(items.at(0).prop('city')).toBe(props.cities[0]);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call checkIfFeatured', () => {
    shallow(<CitiesList {...props} />);

    expect(props.checkIfFeatured).toBeCalledTimes(2);
    expect(props.checkIfFeatured).toHaveBeenNthCalledWith(1, 0);
    expect(props.checkIfFeatured).toHaveBeenNthCalledWith(2, 1);
  });
});
