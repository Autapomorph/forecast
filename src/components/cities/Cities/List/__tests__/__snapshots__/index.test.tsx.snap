// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CitiesList tests renders properly 1`] = `
<styles__CitiesList>
  <CitiesItem
    addToFeatured={[MockFunction]}
    city={
      Object {
        "coords": Object {
          "latitude": 0,
          "longitude": 0,
        },
        "countryCode": "countryCode",
        "countryName": "countryName",
        "id": 0,
        "name": "name",
        "region": "region",
      }
    }
    fetchCity={[MockFunction]}
    key="0"
    removeFromFeatured={[MockFunction]}
  />
  <CitiesItem
    addToFeatured={[MockFunction]}
    city={
      Object {
        "coords": Object {
          "latitude": 0,
          "longitude": 0,
        },
        "countryCode": "countryCode2",
        "countryName": "countryName2",
        "id": 1,
        "name": "name2",
        "region": "region2",
      }
    }
    fetchCity={[MockFunction]}
    key="1"
    removeFromFeatured={[MockFunction]}
  />
</styles__CitiesList>
`;

exports[`CitiesList tests should render CitiesItem 1`] = `
<styles__CitiesList>
  <CitiesItem
    addToFeatured={[MockFunction]}
    city={
      Object {
        "coords": Object {
          "latitude": 0,
          "longitude": 0,
        },
        "countryCode": "countryCode",
        "countryName": "countryName",
        "id": 0,
        "name": "name",
        "region": "region",
      }
    }
    fetchCity={[MockFunction]}
    key="0"
    removeFromFeatured={[MockFunction]}
  />
  <CitiesItem
    addToFeatured={[MockFunction]}
    city={
      Object {
        "coords": Object {
          "latitude": 0,
          "longitude": 0,
        },
        "countryCode": "countryCode2",
        "countryName": "countryName2",
        "id": 1,
        "name": "name2",
        "region": "region2",
      }
    }
    fetchCity={[MockFunction]}
    key="1"
    removeFromFeatured={[MockFunction]}
  />
</styles__CitiesList>
`;
